kube-runtime/src/watcher.rs

```diff
@@ -48,28 +48,24 @@ pub enum Event<K> {
     Delete(K),
     /// The watch stream was restarted.
     ///
-    /// If using the `ListWatch` strategy, `RestartPage` events will follow this event.
-    /// If using the `StreamingList` strategy, this event will be followed by `RestartApply` or `RestartDelete` events.
-    RestartInit,
-    /// A page of objects was received during the restart.
+    /// If using the `ListWatch` strategy, this is a relist, and indicates that `InitlPage` events will follow.
+    /// If using the `StreamingList` strategy, this event will be followed by `InitApply` + `InitDelete` events.
+    Init,
```

Comment:
Do we actually gain much from having this be a separate variant, or would it be enough to say that `InitPage`/`InitApply`/`Ready` implies the start of an init if one is not already happening?

Reply:
You could do it either way, but it feels slightly cleaner to have an event for it where you can clear any temporary buffer. Otherwise you'd have to check if it's the first page, and only do it then in an `if` - and you have to do it in two places if you need to support both strategies (since one gives a vec and the other gives an item).

Reply:
I think it works out kind of naturally to call `buffer.or_insert_default()` in `InitPage` and `buffer.take()` in `InitReady` (where `buffer: Option<_>`), but it's a fair point.